src/ompl/util/RandomNumbers.h

```diff
@@ -43,8 +43,18 @@
 #include <boost/random/variate_generator.hpp>
 #include <cassert>
 
+#include <ompl/util/ClassForward.h>
+
+//The basic Eigen include
+#include <Eigen/Core>
```

Comment:
So, I tried, but it's complicated.
Eigen is heavily templated, and the two Eigen classes used are typedef'ed versions of the class: 

    template<typename _Scalar, int _Rows, int _Cols, int _Options, int _MaxRows, int _MaxCols>
   Eigen::Matrix< _Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols >

As most of the template arguments are "magic numbers" or options (i.e., _Rows is set to `Dynamic` for `Eigen::MatrixXd`), I didn't know how I could cleanly do PIMPL it. I could copy the value of those magic numbers (i.e., `Dynamic` is -1), but that seems really fragile.